main.ml

```diff
@@ -354,17 +365,21 @@ module Datapath_plugins = struct
   let table = String.Table.create ()
 
   let register ~datapath_root datapath_plugin_name =
-    let script_dir = Filename.concat datapath_root datapath_plugin_name in
-    return_plugin_rpc (fun () -> Plugin_client.query (fork_exec_rpc ~script_dir) "register")
-    >>= fun result ->
-    match result with
-    | Ok response ->
-      info "Registered datapath plugin %s" datapath_plugin_name;
-      Hashtbl.set table ~key:datapath_plugin_name ~data:(script_dir, response);
-      return ()
-    | Error _ ->
-      info "Failed to register datapath plugin %s" datapath_plugin_name;
-      return ()
+    let result =
+      let script_dir = Filename.concat datapath_root datapath_plugin_name in
+      return_plugin_rpc (fun () -> Plugin_client.query (fork_exec_rpc ~script_dir) "register")
+      >>>= fun response ->
+      check_plugin_version_compatible response >>= function
+      | Ok () ->
+        info "Registered datapath plugin %s" datapath_plugin_name;
+        Hashtbl.set table ~key:datapath_plugin_name ~data:(script_dir, response);
+        return (Ok ())
+      | Error e ->
+        let err_msg = Storage_interface.Exception.rpc_of_exnty e |> Jsonrpc.to_string in
+        info "Failed to register datapath plugin %s: %s" datapath_plugin_name err_msg;
+        return (Error e)
+    in
+    result >>= fun _ -> return ()
```

Comment:
Why do you need this extra ignore, doesn't the inner one already return a unit deferred?

Reply:
Because we return a `result` and use the `bind` for `result`s up there, so we have to get outside the result bind to return a simple `unit`.

Reply:
Maybe add a comment saying that we ignore errors here: we just do not register the plugin if we've encountered any error. In the future we might want to change that, so we keep the error result here.

Reply:
Sounds good, done.